nautobot_device_lifecycle_mgmt/metrics.py

```diff
@@ -0,0 +1,57 @@
+"""Nautobot Device LCM  plugin application level metrics ."""
+from datetime import datetime
+
+from django.conf import settings
+from nautobot.dcim.models import Device, Site, InventoryItem
+from prometheus_client.core import GaugeMetricFamily
+
+from nautobot_device_lifecycle_mgmt.models import HardwareLCM
+
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_device_lifecycle_mgmt", {})
+
+
+def nautobot_metrics_dlcm_eos():
+    """Calculate number of EOS devices per Device Type and per Site.
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    current_dt = datetime.now()
+    hw_eos_notices = HardwareLCM.objects.filter(end_of_support__lte=current_dt)
+    hw_eos_device_types = [notice.device_type for notice in hw_eos_notices]
+    hw_eos_inventoryitems = [notice.inventory_item for notice in hw_eos_notices]
+
+    part_number_gauge = GaugeMetricFamily(
+        "nautobot_lcm_devices_eos_per_part_number", "Nautobot LCM Devices EOS per Part Number", labels=["part_number"]
+    )
+    devices_gauge = GaugeMetricFamily(
+        "nautobot_lcm_devices_eos_per_site", "Nautobot LCM Devices EOS per Site", labels=["site"]
+    )
+
+    for notice in hw_eos_notices:
+        if notice.device_type:
+            part_number = notice.device_type.part_number
+            eos_devices = Device.objects.filter(device_type=notice.device_type)
+            metric_value = eos_devices.count()
+        elif notice.inventory_item:
+            part_number = notice.inventory_item
+            eos_devices = InventoryItem.objects.filter(part_id=notice.inventory_item)
+            metric_value = eos_devices.count()
+        else:
+            continue
+
+        part_number_gauge.add_metric(labels=[part_number], value=metric_value)
```

Comment:
```suggestion
            eos_devices_count = Device.objects.filter(device_type=notice.device_type).count()
        elif notice.inventory_item:
            part_number = notice.inventory_item
            eos_devices_count = InventoryItem.objects.filter(part_id=notice.inventory_item).count()
        else:
            continue

        part_number_gauge.add_metric(labels=[part_number], value=eos_devices_count)
```

Reply:
@nkallergis what is the goal of this item?

Reply:
Sorry for this, bad comment on my side on multiple fronts! I intended to
1.  Remove the `metric_value = eos_devices.count()` lines by moving `.count()` to the previous ones. Not because one approach is _better_ than the other, but because you're using one-liners in lines 47-50 and it would be nice to be consistent.
1. Remove the `else-continue` clause - this shouldn't be required as either a Device Type or an Inventory Item is mandatory for a HW Notice, right?